superdesk/io/rfc822.py

```diff
@@ -304,3 +308,96 @@ def safe_css(self, attr, css):
+        The google form submits to a google sheet, this sheet creates the email as a notification
+        :param data:
+        :param provider:
+        :return: A list of 1 item
+        """
+        try:
+            item = dict()
+            item[ITEM_TYPE] = CONTENT_TYPE.TEXT
+            item['versioncreated'] = utcnow()
+            for response_part in data:
+                if isinstance(response_part, tuple):
+                    msg = email.message_from_bytes(response_part[1])
+                    item['guid'] = msg['Message-ID']
+                    date_tuple = email.utils.parsedate_tz(msg['Date'])
+                    if date_tuple:
+                        dt = datetime.datetime.utcfromtimestamp(
+                            email.utils.mktime_tz(date_tuple))
+                        dt = dt.replace(tzinfo=timezone('utc'))
+                        item['firstcreated'] = dt
+
+                    for part in msg.walk():
+                        if part.get_content_type() == "text/plain":
+                            body = part.get_payload(decode=True)
+                            # if we don't know the charset just have a go!
+                            if part.get_content_charset() is None:
+                                json_str = body.decode().replace('\r\n', '').replace('  ', ' ')
+                            else:
+                                charset = part.get_content_charset()
+                                json_str = body.decode(charset).replace('\r\n', '').replace('  ', ' ')
+
+                            mail_item = dict((k, v[0]) for k, v in json.loads(json_str).items())
+
+                            self._expand_category(item, mail_item)
+
+                            item['original_source'] = mail_item.get('Username')
+                            item['headline'] = mail_item.get('Headline')
+                            item['abstract'] = mail_item.get('Abstract')
+                            item['slugline'] = mail_item.get('Slugline')
+                            item['body_html'] = mail_item.get('Body')
+                            item[SIGN_OFF] = mail_item.get('Sign off')
+
+                            if mail_item.get('Priority') != '':
+                                item['priority'] = int(mail_item.get('Priority'))
+                            if mail_item.get('Urgency') != '':
+                                item['urgency'] = int(mail_item.get('Urgency'))
+
+                            # We expect the username passed coresponds to a superdesk user
+                            lookup = superdesk.get_resource_service('users').find_one(
+                                req=ParsedRequest(), email=item['original_source'])
+                            if not lookup:
+                                raise UserNotRegisteredException()
+                            item['original_creator'] = lookup.get('_id')
+                            item['byline'] = lookup.get('display_name')
```

Comment:
This will ignore the byline if it's been set in user profile. should be `lookup.get('byline', lookup.get('display_name'))`.